pkg/controller/pipelines.go

```diff
@@ -59,22 +52,57 @@ func (c *Controller) PullRefMetrics(ctx context.Context, ref schemas.Ref) error
 		return nil
 	}
 
-	pipeline, err := c.Gitlab.GetRefPipeline(ctx, ref, pipelines[0].ID)
+	slices.Reverse(pipelines)
```

Comment:
why is this reversed ?

Reply:
This is also to make the looping code as backwards compatible to the vanilla version of the code as possible.
The `ProcessPipelineMetrics` method has the side effect to store a `LatestPipeline` in the `ref`. By reversing we're processing the first one last giving us the same `LatestPipeline` as the vanilla code. This is more a precaution to not break any assumptions about the `LatestPipeline`. Other than that the order of the processed pipelines should not matter all that much.
Maybe I can leave a comment explaining this in the code. After a week I also needed a hard think why that was needed.

Reply:
The tests would be green even without this line, but as I said I leave it more as a precaution to not break assumptions that the original author has on the `ref.LatestPipeline`.